src/QsCompiler/LanguageServer/LanguageServer.cs

```diff
@@ -764,6 +764,7 @@ public object OnCodeAction(JToken arg)
                 return
                     param.Command == CommandIds.ApplyEdit ? CastAndExecute<ApplyWorkspaceEditParams>(edit =>
                         this.rpc.InvokeWithParameterObjectAsync<ApplyWorkspaceEditResponse>(Methods.WorkspaceApplyEditName, edit)) :
+                    param.Command == CommandIds.FileIsNotebookCell ? CastAndExecute<TextDocumentIdentifier>(this.editorState.FileIsNotebookCell) :
```

Comment:
I can't decide if this is goofy or not. The code to get this to work is pretty ugly thanks to [C# limitations][1], and I'm not sure it contributes much to the unit tests where it's used. But it took me long enough to get it to work that I included it anyway

[1]: https://devblogs.microsoft.com/dotnet/try-out-nullable-reference-types/#the-issue-with-t

Reply:
I think it's ok to have.

Reply:
After re-plumbing DocumentKind everywhere, it feels less disturbing thankfully